pkg/proxy/handler/query.go

```diff
@@ -35,14 +35,22 @@ func HandleLokiQueries(w http.ResponseWriter, results <-chan *http.Response, log
 
 		// Decode into map[string]interface{} to inspect the raw structure
 		var rawBody map[string]interface{}
-		if err := json.Unmarshal(bodyBytes, &rawBody); err != nil {
-			level.Error(logger).Log("msg", "Failed to decode raw JSON", "err", err)
-			continue
+		if json.Valid(bodyBytes) {
+			if err := json.Unmarshal(bodyBytes, &rawBody); err != nil {
+				level.Error(logger).Log("msg", "Failed to decode raw JSON", "err", err)
+			} else {
+				rawBodyJSON, err := json.Marshal(rawBody)
```

Comment:
Hi, does it make sense to re-marshal just for logging, since previously you unmarshal? Maybe something like this would work?

```suggestion
var rawBody map[string]interface{}
bodyStr := string(bodyBytes)
if json.Valid(bodyBytes) {
    if err := json.Unmarshal(bodyBytes, &rawBody); err != nil {
        level.Error(logger).Log("msg", "Failed to decode JSON", "err", err)
    } else {
        level.Debug(logger).Log("msg", "Raw JSON body", "rawBody", bodyStr)
    }
} else {
    level.Debug(logger).Log("msg", "Raw body is not JSON", "rawBody", bodyStr)
}
```

Reply:
yes, makes sense. pushed